internal/services/network/subnet_resource.go

```diff
@@ -349,6 +356,10 @@ func resourceSubnetCreate(d *pluginsdk.ResourceData, meta interface{}) error {
 			privateLinkServiceNetworkPoliciesRaw = d.Get("private_link_service_network_policies_enabled").(bool)
 		}
 
+		if !pluginsdk.IsExplicitlyNullInConfig(d, "default_outbound_access_enabled") {
+			properties.DefaultOutboundAccess = pointer.To(d.Get("default_outbound_access_enabled").(bool))
+		}
```

Comment:
Which means this can become a regular assignment:

```suggestion
		properties.DefaultOutboundAccess = pointer.To(d.Get("default_outbound_access_enabled").(bool))
```